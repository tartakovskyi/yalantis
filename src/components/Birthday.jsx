import React  from 'react'


const Birthday = ({ user:{dob, firstName, lastName} }) => {

  return (
    <li>{firstName} {lastName}</li>
  )
}


export default Birthday