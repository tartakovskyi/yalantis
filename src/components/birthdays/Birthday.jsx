import React from "react";
import PropTypes from "prop-types";

const Birthday = ({ user: { dob, firstName, lastName }, month }) => {
  const DateString = () => {
    const d = new Date(dob);
    return ("0" + d.getDate()).slice(-2) + " " + month + " " + d.getFullYear();
  };

  return (
    <li>
      {firstName} {lastName} - <DateString />
    </li>
  );
};

Birthday.propTypes = {
  user: PropTypes.shape({
    dob: PropTypes.string.isRequired,
    firstName: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired,
  }).isRequired,
  month: PropTypes.string.isRequired,
};

export default React.memo(Birthday);
