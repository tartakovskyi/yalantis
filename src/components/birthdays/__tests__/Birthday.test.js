import React from "react";
import { render } from "@testing-library/react";
import Birthday from "../Birthday";

const user = {
  id: "5e00928df3f5ec0978d90ce7",
  firstName: "Madeleine",
  lastName: "Brooks",
  dob: "2019-05-12T01:20:34.085Z",
};

test("Birthday should render correctly", () => {
  const { container, getAllByTestId } = render(
    <Birthday month="May" user={user} />
  );

  expect(container.firstChild).toMatchSnapshot();
});
