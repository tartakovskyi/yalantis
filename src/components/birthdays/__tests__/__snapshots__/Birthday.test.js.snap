// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Birthday should render correctly 1`] = `
<li>
  Madeleine
   
  Brooks
   - 
  12 May 2019
</li>
`;
